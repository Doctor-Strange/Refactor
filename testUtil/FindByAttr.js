

export const FindByAttr = (wrapper, value) => {
  return wrapper.find(`[data-test="${value}"]`);
};
