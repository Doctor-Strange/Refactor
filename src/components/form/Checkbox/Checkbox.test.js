import React from "react";
import { mount } from "enzyme";
import { FindByAttr } from "../../../../testUtil/FindByAttr";
import CheckBox from "./";

const clearFieldMock = jest.fn();
const SelectMock = jest.fn();

const props = {
  data: [
    { value: "a", text: "a" },
    { value: "b", text: "b" },
    { value: "c", text: "c" },
  ],
  name: "name",
  clearField: clearFieldMock,
  Select: SelectMock,
};
const setup = (props) => {
  return mount(<CheckBox {...props} />);
};

describe("Checkbox component", () => {
  describe("snapshot", () => {
    test("/component/form/checkbox snapshot", () => {
      const wrapper = setup(props);
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe("show spinner", () => {
    test("empty data should show spinner", () => {
      const wrapper = setup({ ...props, data: [] });
      const resultList = FindByAttr(wrapper, "resultList");
      expect(resultList.length).toEqual(1);
    });
  });
  describe("show checkboxes", () => {
    test("render 3 checkboxes", () => {
      const wrapper = setup({ ...props });
      const container = FindByAttr(wrapper, "container");
      expect(container.length).toEqual(3);
      expect(container.first().text()).toEqual("a");
    });
  });
  describe("change controller", () => {
    const wrapper = setup({ ...props });
    const checkbox = FindByAttr(wrapper, "checkbox");
    test("click and call `SelectMock`", () => {
      checkbox.first().simulate("change", { target: { checked: true } });
      expect(SelectMock).toHaveBeenCalledTimes(1);
      expect(SelectMock).toHaveBeenCalledWith({ value: "a", text: "a" });
    });
    test("click and call `SelectMock`", () => {
      checkbox.first().simulate("change", { target: { checked: false } });
      expect(clearFieldMock).toHaveBeenCalledTimes(1);
      expect(clearFieldMock).toHaveBeenCalledWith({ value: "a", text: "a" });
    });
  });
});
