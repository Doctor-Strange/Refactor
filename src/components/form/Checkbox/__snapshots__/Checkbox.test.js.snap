// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component snapshot /component/form/checkbox snapshot 1`] = `ReactWrapper {}`;
